gdal: 3.9.3 -> 3.10.0

khaneliman opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done
Follow up to https://github.com/NixOS/nixpkgs/pull/354783

https://github.com/OSGeo/gdal/releases/tag/v3.10.0
I see some new optional dependencies reported:

    New optional dependencies
    [libavif](https://github.com/AOMediaCodec/libavif) for AVIF driver
    [libopendrive](https://github.com/DLR-TS/libOpenDRIVE) for XODR driver

Added libavif optional dependency, but don't see libopendrive packaged.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

khaneliman committed:
gdal: 3.9.3 -> 3.10.0

khaneliman committed:
gdal: add libavif optional dependency

autra commented:
Thanks to have splitted that. I'd vote to wait for ZHF to end indeed, as gdal tends to break other packages even with minor versions (usually it's just some tests, but let's stay on the safe side of things).

emilazy committed:
pdal: add upstream patch for GDAL 3.10

emilazy committed:
python3Packages.rasterio: 1.4.0 -> 1.4.2

Diff: https://github.com/rasterio/rasterio/compare/refs/tags/1.4.0...1.4.2

Changelog: https://github.com/rasterio/rasterio/blob/1.4.2/CHANGES.txt

emilazy committed:
python3Packages.rio-tiler: 7.0.1 -> 7.2.2

Diff: https://github.com/cogeotiff/rio-tiler/compare/refs/tags/7.0.1...7.2.2

emilazy commented:
@NixOS/geospatial Hi there, `rio-tiler`’s tests seem to be segfaulting on this branch. We should move on this update soon because the GCC 14 update is going to break these packages, so it’d be great if you could take a look. I’m happy to help in any way I can.

emilazy commented:
Do you think we should just mark `rio-tiler` broken for now, or is it too important? I’m not sure where to start debugging the segfaulting test.

imincik commented:
> Do you think we should just mark `rio-tiler` broken for now, or is it too important? I’m not sure where to start debugging the segfaulting test.

rio-tiler is quite important, but I think we can just submit bug report upstream.

imincik commented:
https://github.com/cogeotiff/rio-tiler/issues/769

imincik commented:
> [cogeotiff/rio-tiler#769](https://github.com/cogeotiff/rio-tiler/issues/769)

If we can't provide a fix we should mark rio-tiler as broken and unblock this GDAL update.

khaneliman committed:
python3Packages.rio-tiler: mark broken

https://github.com/cogeotiff/rio-tiler/issues/769

imincik commented:
Result of `nixpkgs-review pr 355220` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.python-mapnik</li>
    <li>python311Packages.python-mapnik.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python311Packages.worldengine</li>
    <li>python311Packages.worldengine.dist</li>
    <li>python312Packages.python-mapnik</li>
    <li>python312Packages.python-mapnik.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.worldengine</li>
    <li>python312Packages.worldengine.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>apacheHttpdPackages.mod_tile</li>
    <li>t-rex</li>
  </ul>
</details>
<details>
  <summary>157 packages built:</summary>
  <ul>
    <li>cloudcompare</li>
    <li>entwine</li>
    <li>gdal (python312Packages.gdal)</li>
    <li>gdalMinimal</li>
    <li>gmt</li>
    <li>gplates</li>
    <li>grass</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapproxy.dist</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>mysql-workbench</li>
    <li>octavePackages.mapping</li>
    <li>openorienteering-mapper</li>
    <li>paraview</li>
    <li>pdal</li>
    <li>perl538Packages.Tirex</li>
    <li>perl538Packages.Tirex.devdoc</li>
    <li>perl540Packages.Tirex</li>
    <li>perl540Packages.Tirex.devdoc</li>
    <li>postgresql13JitPackages.postgis</li>
    <li>postgresql13JitPackages.postgis.doc</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresql13Packages.postgis.doc</li>
    <li>postgresql14JitPackages.postgis</li>
    <li>postgresql14JitPackages.postgis.doc</li>
    <li>postgresql14Packages.postgis</li>
    <li>postgresql14Packages.postgis.doc</li>
    <li>postgresql15JitPackages.postgis</li>
    <li>postgresql15JitPackages.postgis.doc</li>
    <li>postgresql15Packages.postgis</li>
    <li>postgresql15Packages.postgis.doc</li>
    <li>postgresql16JitPackages.postgis</li>
    <li>postgresql16JitPackages.postgis.doc</li>
    <li>postgresql16Packages.postgis</li>
    <li>postgresql16Packages.postgis.doc</li>
    <li>postgresql17JitPackages.postgis</li>
    <li>postgresql17JitPackages.postgis.doc</li>
    <li>postgresql17Packages.postgis</li>
    <li>postgresql17Packages.postgis.doc</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.cartopy</li>
    <li>python311Packages.cartopy.dist</li>
    <li>python311Packages.django-bootstrap4</li>
    <li>python311Packages.django-bootstrap4.dist</li>
    <li>python311Packages.django-bootstrap5</li>
    <li>python311Packages.django-bootstrap5.dist</li>
    <li>python311Packages.drf-extra-fields</li>
    <li>python311Packages.drf-extra-fields.dist</li>
    <li>python311Packages.fiona</li>
    <li>python311Packages.fiona.dist</li>
    <li>python311Packages.folium</li>
    <li>python311Packages.folium.dist</li>
    <li>python311Packages.gdal</li>
    <li>python311Packages.geoarrow-pandas</li>
    <li>python311Packages.geoarrow-pandas.dist</li>
    <li>python311Packages.geoarrow-pyarrow</li>
    <li>python311Packages.geoarrow-pyarrow.dist</li>
    <li>python311Packages.geodatasets</li>
    <li>python311Packages.geodatasets.dist</li>
    <li>python311Packages.geopandas</li>
    <li>python311Packages.geopandas.dist</li>
    <li>python311Packages.geoparquet</li>
    <li>python311Packages.geoparquet.dist</li>
    <li>python311Packages.inequality</li>
    <li>python311Packages.inequality.dist</li>
    <li>python311Packages.libpysal</li>
    <li>python311Packages.libpysal.dist</li>
    <li>python311Packages.mapclassify</li>
    <li>python311Packages.mapclassify.dist</li>
    <li>python311Packages.momepy</li>
    <li>python311Packages.momepy.dist</li>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.netbox-documents</li>
    <li>python311Packages.netbox-documents.dist</li>
    <li>python311Packages.osmnx</li>
    <li>python311Packages.osmnx.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.pygmt</li>
    <li>python311Packages.pygmt.dist</li>
    <li>python311Packages.pyogrio</li>
    <li>python311Packages.pyogrio.dist</li>
    <li>python311Packages.rasterio</li>
    <li>python311Packages.rasterio.dist</li>
    <li>python311Packages.rioxarray</li>
    <li>python311Packages.rioxarray.dist</li>
    <li>python311Packages.shimmy</li>
    <li>python311Packages.shimmy.dist</li>
    <li>python311Packages.wktutils</li>
    <li>python311Packages.wktutils.dist</li>
    <li>python312Packages.bsuite</li>
    <li>python312Packages.bsuite.dist</li>
    <li>python312Packages.cartopy</li>
    <li>python312Packages.cartopy.dist</li>
    <li>python312Packages.django-bootstrap4</li>
    <li>python312Packages.django-bootstrap4.dist</li>
    <li>python312Packages.django-bootstrap5</li>
    <li>python312Packages.django-bootstrap5.dist</li>
    <li>python312Packages.drf-extra-fields</li>
    <li>python312Packages.drf-extra-fields.dist</li>
    <li>python312Packages.fiona</li>
    <li>python312Packages.fiona.dist</li>
    <li>python312Packages.folium</li>
    <li>python312Packages.folium.dist</li>
    <li>python312Packages.geoarrow-pandas</li>
    <li>python312Packages.geoarrow-pandas.dist</li>
    <li>python312Packages.geoarrow-pyarrow</li>
    <li>python312Packages.geoarrow-pyarrow.dist</li>
    <li>python312Packages.geodatasets</li>
    <li>python312Packages.geodatasets.dist</li>
    <li>python312Packages.geopandas</li>
    <li>python312Packages.geopandas.dist</li>
    <li>python312Packages.geoparquet</li>
    <li>python312Packages.geoparquet.dist</li>
    <li>python312Packages.inequality</li>
    <li>python312Packages.inequality.dist</li>
    <li>python312Packages.libpysal</li>
    <li>python312Packages.libpysal.dist</li>
    <li>python312Packages.mapclassify</li>
    <li>python312Packages.mapclassify.dist</li>
    <li>python312Packages.momepy</li>
    <li>python312Packages.momepy.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.osmnx</li>
    <li>python312Packages.osmnx.dist</li>
    <li>python312Packages.plotnine</li>
    <li>python312Packages.plotnine.dist</li>
    <li>python312Packages.pygmt</li>
    <li>python312Packages.pygmt.dist</li>
    <li>python312Packages.pyogrio</li>
    <li>python312Packages.pyogrio.dist</li>
    <li>python312Packages.rasterio</li>
    <li>python312Packages.rasterio.dist</li>
    <li>python312Packages.rioxarray</li>
    <li>python312Packages.rioxarray.dist</li>
    <li>python312Packages.wktutils</li>
    <li>python312Packages.wktutils.dist</li>
    <li>pytrainer</li>
    <li>pytrainer.dist</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
    <li>qmapshack</li>
    <li>saga</li>
    <li>sumo</li>
    <li>survex</li>
    <li>therion</li>
    <li>tunnelx</li>
    <li>udig</li>
    <li>vpv</li>
  </ul>
</details>

imincik reviewed (approved):
Looks good to me. Thank you very much @khaneliman !
